### PR TITLE
Delete user account

### PR DESCRIPTION
**Pull request by creativeappdevamit**
# What and why?

# How to test?

# Trello
https://trello.com/c/h8nV6an7/883-user-account-creation-account-deletion-request

### CODE REVIEW

File: frontstage/controllers/auth_controller.py [hunk 1]
```diff
@@ -51,3 +52,23 @@ def sign_in(username, password):
 
     bound_logger.info("Successfully signed in")
     return {}
+
+
+def delete_account(username: str):
+    bound_logger = logger.bind(email=obfuscate_email(username))
+    bound_logger.info("Attempting to delete account")
+    auth = "{}:{}".format(app.config["SECURITY_USER_NAME"], app.config["SECURITY_USER_PASSWORD"]).encode("utf-8")
```
Comment (insacuri) [inline frontstage/controllers/auth_controller.py:60]: Why bother with all this faff?  All the other ones do something like 
```python
response = requests.delete(url, data=form_data, auth=app.config["BASIC_AUTH"])
```
without needing to spend so much time making auth headers.  Or is there something I'm missing?

Comment (creativeappdevamit) [reply frontstage/controllers/auth_controller.py:60]: no may be you are right ... I just used the way it was used previously

Comment (creativeappdevamit) [reply frontstage/controllers/auth_controller.py]: done!

File: frontstage/views/account/account_delete.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+from frontstage.views.account import account_bp
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+@account_bp.route("/delete", methods=["GET", "POST"])
+@jwt_authorization(request)
+def delete_user_account(session):
+    party_id = session.get_party_id()
+    respondent_details = party_controller.get_respondent_party_by_id(party_id)
```
Comment (insacuri) [inline frontstage/views/account/account_delete.py:19]: Thinking about it.  There should be a final check somewhere in this function to make sure they really don't have any surveys attached to them.
Nothing has been done to stop me knowing this delete endpoint exists, going to it manually via url and pressing the delete button.

File: frontstage/views/sign_in/sign_in.py [hunk 3]
```diff
@@ -70,6 +71,8 @@ def login():  # noqa: C901
                 bound_logger.info("Bad credentials provided")
             elif UNKNOWN_ACCOUNT_ERROR in error_message:
                 bound_logger.info("User account does not exist in auth service")
+            elif USER_ACCOUNT_DELETED in error_message:
+                bound_logger.info("User account is marked for deletion")
```
Comment (insacuri) [inline frontstage/views/sign_in/sign_in.py:75]: Should this be something like:
```python
bound_logger.info("User account is marked for deletion.  Contact us if you believe this has been marked in error")
```
or something like that?  Might need to ask Gareth what he thinks.  I could also be wrong so ignore this if you want ^^

**Review by insacuri — CHANGES_REQUESTED** [review]
Mostly looking good.  I just want a bit more validation just before the deletion to prevent misuse

File: frontstage/views/account/account_delete.py (hunk 2 above)
Comment (creativeappdevamit) [reply frontstage/views/account/account_delete.py:19]: 👍

File: frontstage/views/sign_in/sign_in.py (hunk 3 above)
Comment (creativeappdevamit) [reply frontstage/views/sign_in/sign_in.py:75]: 👍

Comment (creativeappdevamit) [reply frontstage/views/sign_in/sign_in.py:75]: Although its just a logger

Comment (creativeappdevamit) [reply frontstage/views/sign_in/sign_in.py:75]: checked with Gareth. He is happy as it is

Comment (insacuri) [reply frontstage/views/sign_in/sign_in.py:75]: I'm dumb, it is just a logger.  Sorry >_<

**Review by insacuri — APPROVED** [review]

**Pull request merged**